#1 feat : 프로젝트 생성

uri010 opened the pull request:
jj

uri010 committed:
#1 feat : 프로젝트 생성

xeonu commented on build.gradle:
```diff
@@ -0,0 +1,31 @@
+}
+
+dependencies {
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	implementation 'org.mybatis.spring.boot:mybatis-spring-boot-starter:2.3.0'
+	compileOnly 'org.projectlombok:lombok'
+	annotationProcessor 'org.projectlombok:lombok'
+	testImplementation 'org.springframework.boot:spring-boot-starter-test'
```
implemntation, compileOnly, annotationProcessor, testImplementation이 무엇을 의미하나요?

xeonu commented on HELP.md:
```diff
@@ -0,0 +1,29 @@
+# Read Me First
+The following was discovered as part of building this project:
+
+* The original package name 'study.shared-calendar' is invalid and this project uses 'study.sharedcalendar' instead.
+
+# Getting Started
+
+### Reference Documentation
+For further reference, please consider the following sections:
+
+* [Official Gradle documentation](https://docs.gradle.org)
+* [Spring Boot Gradle Plugin Reference Guide](https://docs.spring.io/spring-boot/docs/2.7.8/gradle-plugin/reference/html/)
+* [Create an OCI image](https://docs.spring.io/spring-boot/docs/2.7.8/gradle-plugin/reference/html/#build-image)
+* [MyBatis Framework](https://mybatis.org/spring-boot-starter/mybatis-spring-boot-autoconfigure/)
+* [Spring Web](https://docs.spring.io/spring-boot/docs/2.7.8/reference/htmlsingle/#web)
+
+### Guides
+The following guides illustrate how to use some features concretely:
+
+* [MyBatis Quick Start](https://github.com/mybatis/spring-boot-starter/wiki/Quick-Start)
+* [Building a RESTful Web Service](https://spring.io/guides/gs/rest-service/)
+* [Serving Web Content with Spring MVC](https://spring.io/guides/gs/serving-web-content/)
+* [Building REST services with Spring](https://spring.io/guides/tutorials/rest/)
+
+### Additional Links
+These additional references should also help you:
+
+* [Gradle Build Scans – insights for your project's build](https://scans.gradle.com#gradle)
+
```
이 파일은 우리 프로젝트에서 어떻게 사용되나요?

xeonu commented on gradlew:
```diff
@@ -0,0 +1,240 @@
+
+set -- \
+        "-Dorg.gradle.appname=$APP_BASE_NAME" \
+        -classpath "$CLASSPATH" \
+        org.gradle.wrapper.GradleWrapperMain \
+        "$@"
+
+# Stop when "xargs" is not available.
+if ! command -v xargs >/dev/null 2>&1
+then
+    die "xargs is not available"
+fi
+
+# Use "xargs" to parse quoted args.
+#
+# With -n1 it outputs one arg per line, with the quotes and backslashes removed.
+#
+# In Bash we could simply go:
+#
+#   readarray ARGS < <( xargs -n1 <<<"$var" ) &&
+#   set -- "${ARGS[@]}" "$@"
+#
+# but POSIX shell has neither arrays nor command substitution, so instead we
+# post-process each arg (as a line of input to sed) to backslash-escape any
+# character that might be a shell metacharacter, then use eval to reverse
+# that process (while maintaining the separation between arguments), and wrap
+# the whole thing up as a single "set" statement.
+#
+# This will of course break if any of these variables contains a newline or
+# an unmatched quote.
+#
+
+eval "set -- $(
+        printf '%s\n' "$DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS" |
+        xargs -n1 |
+        sed ' s~[^-[:alnum:]+,./:=@_]~\\&~g; ' |
+        tr '\n' ' '
+    )" '"$@"'
+
+exec "$JAVACMD" "$@"
```
이 파일의 용도는 무엇인가요?

xeonu commented on gradlew.bat:
```diff
@@ -0,0 +1,91 @@
+
+:findJavaFromJavaHome
+set JAVA_HOME=%JAVA_HOME:"=%
+set JAVA_EXE=%JAVA_HOME%/bin/java.exe
+
+if exist "%JAVA_EXE%" goto execute
+
+echo.
+echo ERROR: JAVA_HOME is set to an invalid directory: %JAVA_HOME%
+echo.
+echo Please set the JAVA_HOME variable in your environment to match the
+echo location of your Java installation.
+
+goto fail
+
+:execute
+@rem Setup the command line
+
+set CLASSPATH=%APP_HOME%\gradle\wrapper\gradle-wrapper.jar
+
+
+@rem Execute Gradle
+"%JAVA_EXE%" %DEFAULT_JVM_OPTS% %JAVA_OPTS% %GRADLE_OPTS% "-Dorg.gradle.appname=%APP_BASE_NAME%" -classpath "%CLASSPATH%" org.gradle.wrapper.GradleWrapperMain %*
+
+:end
+@rem End local scope for the variables with windows NT shell
+if %ERRORLEVEL% equ 0 goto mainEnd
+
+:fail
+rem Set variable GRADLE_EXIT_CONSOLE if you need the _script_ return code instead of
+rem the _cmd.exe /c_ return code!
+set EXIT_CODE=%ERRORLEVEL%
+if %EXIT_CODE% equ 0 set EXIT_CODE=1
+if not ""=="%GRADLE_EXIT_CONSOLE%" exit %EXIT_CODE%
+exit /b %EXIT_CODE%
+
+:mainEnd
+if "%OS%"=="Windows_NT" endlocal
+
+:omega
```
이 파일의 용도는 무엇인가요?

xeonu commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1 @@
+
```
applecation.properties에는 비밀번호와 같은 민감한 정보가 포함될 수 있으니 공개하지 않는 것이 좋습니다.

sunwootest replied:
application.properties에는 비밀번호와 같은 민감한 정보가 포함될 수 있으니 공개하면 안될 것 같습니다 :)

xeonu reviewed (commented):
<br>

sunwootest commented on HELP.md:
```diff
@@ -0,0 +1,29 @@
+# Read Me First
+The following was discovered as part of building this project:
+
+* The original package name 'study.shared-calendar' is invalid and this project uses 'study.sharedcalendar' instead.
+
+# Getting Started
+
+### Reference Documentation
+For further reference, please consider the following sections:
+
+* [Official Gradle documentation](https://docs.gradle.org)
+* [Spring Boot Gradle Plugin Reference Guide](https://docs.spring.io/spring-boot/docs/2.7.8/gradle-plugin/reference/html/)
+* [Create an OCI image](https://docs.spring.io/spring-boot/docs/2.7.8/gradle-plugin/reference/html/#build-image)
+* [MyBatis Framework](https://mybatis.org/spring-boot-starter/mybatis-spring-boot-autoconfigure/)
+* [Spring Web](https://docs.spring.io/spring-boot/docs/2.7.8/reference/htmlsingle/#web)
+
+### Guides
+The following guides illustrate how to use some features concretely:
+
+* [MyBatis Quick Start](https://github.com/mybatis/spring-boot-starter/wiki/Quick-Start)
+* [Building a RESTful Web Service](https://spring.io/guides/gs/rest-service/)
+* [Serving Web Content with Spring MVC](https://spring.io/guides/gs/serving-web-content/)
+* [Building REST services with Spring](https://spring.io/guides/tutorials/rest/)
+
+### Additional Links
+These additional references should also help you:
+
+* [Gradle Build Scans – insights for your project's build](https://scans.gradle.com#gradle)
+
```
해당 파일이 꼭 우리 프로젝트에 필요할까요?

uri010 replied:
IntelliJ에서 자동생성해주는 파일로 README.md와 같은 역할이라 저희 프로젝트 상에선 필요 없을 것 같습니다. 삭제하겠습니다.

sunwootest commented on build.gradle:
```diff
@@ -0,0 +1,31 @@
+}
+
+dependencies {
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	implementation 'org.mybatis.spring.boot:mybatis-spring-boot-starter:2.3.0'
+	compileOnly 'org.projectlombok:lombok'
+	annotationProcessor 'org.projectlombok:lombok'
+	testImplementation 'org.springframework.boot:spring-boot-starter-test'
```
implementation, compileOnly, annotationProcessor, testImplementation이 무엇을 의미하나요?

uri010 replied:
- implementation : 해당 의존성을 직접 의존하고 있는 모듈만 재빌드해서 빠르고 API 노출을 막아줌
- compileOnly : 해당 의존성을 컴파일시에만 포함 
- annotationProcessor : lombok 사용시 필수로 추가되어야 하는 구문으로 lombok에서 제공하는 어노테이션을 인식할 수 있게 해줌
- testImplementation : 테스트 코드 수행할 때만 implementation의 기능을 수행

#### 이외의 의존성 옵션들 
- api : 의존 라이브러리 수정시 본 모듈을 의존하는 모듈들도 재빌드 해주는 옵션으로  implementation과 달리 본 모듈을 의존하는 모듈들도 해당 라이브러리의 api를 사용 가능
- runtimeOnly : runtime 시에만 필요한 라이브러리인 경우 사용하는 옵션

이정도로 정리해볼 수 있을 것 같습니다!

sunwootest commented on gradlew:
```diff
@@ -0,0 +1,240 @@
+#!/bin/sh
+
+#
+# Copyright © 2015-2021 the original authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
```
이 파일은 무슨 역할을 하나요?

yunah7 replied:
리눅스, 맥 운영체제 용 실행 쉘 스크립트 파일로 gradle-wrapper.jar 파일을 사용해 gradle task를 실행합니다

uri010 replied:
gradle은 거의 모든 유형의 소프트웨어를 빌드할 수 있을 정도로 유연하게 설계된 오픈소스 빌드 자동화 기구로 JVM에서 실행돼 JDK가 설치되어 있어야 합니다. 매 프로젝트마다 gradle을 설치하는 대신에 gradle wrapper로 새로운 환경에서 프로젝트를 설정할 때 맞는 버전의 java나 gradle을 다시 설치하지 않고 바로 빌드할 수 있게 해줄 수 있습니다. 
gradlew는 맥, 리눅스를 위한 gradle wrapper 파일입니다.

sunwootest commented on gradlew.bat:
```diff
@@ -0,0 +1,91 @@
+@rem
+@rem Copyright 2015 the original author or authors.
+@rem
+@rem Licensed under the Apache License, Version 2.0 (the "License");
+@rem you may not use this file except in compliance with the License.
+@rem You may obtain a copy of the License at
+@rem
+@rem      https://www.apache.org/licenses/LICENSE-2.0
+@rem
```
이 파일은 무슨 역할을 하나요?

uri010 replied:
윈도우를 위한 gradle wrapper 파일입니다.

sunwootest reviewed (commented):
코드리뷰남겼습니다.

sunwootest commented:
리뷰 반영하고 PUSH만 하지 마시고 코멘트 남겨주세요 :)

uri010 committed:
#1 fix : .gitignore 설정 변경, HELP.md 파일 삭제

.gitignore에 불필요한 파일들은 git으로 관리하지 않게 하기 위해서 .gradle, build, .idea 폴더를 추가
이미 README.md 파일이 존재해 HELP.md 파일은 삭제

uri010 commented:
.gitignore 설정 변경 및 HELP.md 파일 삭제했습니다.

uri010 committed:
#1 fix : 원격에 .gitignore 적용

sunwootest reviewed (approved):
approve 드립니다 :)

uri010 committed:
#1 feat : github 이슈,PR 템플릿 추가

uri010 committed:
#1 fix : .github 폴더 삭제
